cmd/cmd.go

```diff
@@ -70,14 +76,32 @@ var rootCmd = &cobra.Command{
 			client = rftp.Client{Conn: rftp.NewUDPConnection()}
 		}
 
-		rs, err := client.Request(hs, files)
+		reqs, err := client.Request(hs, files)
 		if err != nil {
 			log.Printf("error on request: %v\n", err)
 		}
 
-		for _, r := range rs {
-			io.Copy(os.Stdout, r)
-			log.Println("finish write")
+		for i, req := range reqs {
+			var w io.Writer
+			// TODO: verify checksum: method on req?
```

Comment:
TODO seems done? Alternatively, call checksum after rft is done with all bytes and if it doesn't match, add an entry to the error field of the response?